reporter/src/test/kotlin/utils/FreeMarkerTemplateProcessorTest.kt

```diff
@@ -183,35 +215,76 @@ class FreeMarkerTemplateProcessorTest : WordSpec({
-                    locations = emptySet()
-                ),
-                ResolvedLicense(
-                    license = "MIT".toSpdx() as SpdxSingleLicenseExpression,
-                    originalDeclaredLicenses = emptySet(),
-                    originalExpressions = mapOf(LicenseSource.DECLARED to setOf("GPL-2.0-only OR MIT".toSpdx())),
-                    locations = emptySet()
-                )
+    "mergeLicenses" should {
+        "merge the expressions from packages by license sources" {
+            val projects = ORT_RESULT.getProjects().toList()
+            val resolver = mockk<LicenseInfoResolver>()
+            expectResolveLicenseInfo(
+                resolver,
+                projects[0].id,
+                "MIT",
+                mapOf(LicenseSource.DECLARED to setOf("MIT".toSpdx()))
+            )
+            expectResolveLicenseInfo(
+                resolver,
+                projects[1].id,
+                "MIT",
+                mapOf(LicenseSource.DECLARED to setOf("GPL-2.0-only OR MIT".toSpdx()))
             )
+            expectResolveLicenseInfo(
+                resolver,
+                projects[2].id,
+                SpdxConstants.NOASSERTION,
+                mapOf(LicenseSource.DECLARED to setOf(SpdxConstants.NOASSERTION.toSpdx()))
+            )
+            val input = ReporterInput(ORT_RESULT, licenseInfoResolver = resolver)
+            val pkg1 = FreemarkerTemplateProcessor.PackageModel(projects[0].id, input)
+            val pkg2 = FreemarkerTemplateProcessor.PackageModel(projects[1].id, input)
+            val pkg3 = FreemarkerTemplateProcessor.PackageModel(projects[2].id, input)
 
             val result = FreemarkerTemplateProcessor.TemplateHelper(
                 OrtResult.EMPTY,
                 LicenseClassifications(),
                 DefaultResolutionProvider()
-            ).mergeResolvedLicenses(resolvedLicenses)
+            ).mergeLicenses(listOf(pkg1, pkg2, pkg3))
 
+            result should haveSize(2)
             with(result[0]) {
+                license.toString() shouldBe "MIT"
                 originalExpressions[LicenseSource.DECLARED] shouldBe setOf(
                     "MIT".toSpdx(),
                     "GPL-2.0-only OR MIT".toSpdx()
                 )
             }
+
+            result[1].license.toString() shouldBe SpdxConstants.NOASSERTION
+        }
+
+        "filter licenses of excluded packages" {
+            val projects = ORT_RESULT.getProjects().toList()
+            val resolver = mockk<LicenseInfoResolver>()
+            expectResolveLicenseInfo(
+                resolver,
+                projects[0].id,
+                "MIT",
+                mapOf(LicenseSource.DECLARED to setOf("MIT".toSpdx()))
+            )
+
+            val mockResult = mockk<OrtResult>()
+            every { mockResult.isExcluded(any()) } returns false
+            every { mockResult.isExcluded(projects[1].id) } returns true
```

Comment:
It is interesting that this mock call overwrites the former one.

Reply:
It's a specialization, so that if `projects[1].id` is used as the argument something else is returned. Still neat that it works that way 😃

Reply:
I think this is a rather common behavior of mocking frameworks. Also with Mockito, you can define generic default behaviors for mocks and then override them for specific cases.